tests/5.0/program_control/test_capture_omp_affinity.c

```diff
@@ -20,36 +20,39 @@
 
 int main() {
 
-   int i, n, errors, thrd_num;
-   size_t nchars, max_req_store;
+   int i, threads, errors = 0, thrd_num;
+   int  format_size = 13; //size of my_format
+   char format[] = "thrd_num=%0.4n";
 
    char **buffer;
 
+   threads = OMPVV_NUM_THREADS_HOST;
 
-   n = omp_get_num_procs();
-   buffer = (char **)malloc( sizeof(char *) * n );
-   for (i = 0; i < n; i++) { 
-      buffer[i]=(char *)malloc( sizeof(char) * BUFFER_STORE); 
+   buffer = (char **) malloc( sizeof(char *) * threads );
+
+   for (i = 0; i < threads; i++) { 
+      buffer[i] = (char *) malloc( sizeof(char) * BUFFER_STORE); 
    }
-   errors = 0;
-   max_req_store = 0;
 
-   #pragma omp parallel private(thrd_num,nchars) reduction(max:max_req_store)
+   #pragma omp parallel private(thrd_num) num_threads(threads) 
    {
-      if(omp_get_num_threads()>n) exit(1); //safety: dont exceed # of buffers
-
-      thrd_num=omp_get_thread_num();
-      nchars=omp_capture_affinity(buffer[thrd_num],(size_t)BUFFER_STORE,NULL);
+      if(omp_get_num_threads() > threads) exit(1); //safety: dont exceed # of buffers
```

Comment:
Add a msg here before exiting.

Reply:
I removed this line, because it makes no sense to check if the number of threads is greater than the requested, since they have to be less or equal to the num_threads requested.
Instead, I added a variable that checks for the number of threads and only verifies on this number.

Reply:
Exactly, if the number of threads is greater than the requested, then it is an error. I would keep it.

Reply:
Updated the test with the check and the message